src/Components/Utils/PasswordValidation.tsx

```diff
@@ -0,0 +1,20 @@
+export const PasswordValidation = (email: string, password: string, name: string) => {
+  const isValidEmail = /[\w.-]+@([\w-]+\.)+[\w-]{2,4}$/.test(email)
+  const isValidPassword = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/.test(password)
+  const isValidName =
+    /(^[A-Za-z]{3,16})([ ]{0,1})([A-Za-z]{3,16})?([ ]{0,1})?([A-Za-z]{3,16})?([ ]{0,1})?([A-Za-z]{3,16})/.test(
+      name,
+    )
```

Comment:
does not follow Single responsibility principal , 
paraphrase convert it to a hook instead